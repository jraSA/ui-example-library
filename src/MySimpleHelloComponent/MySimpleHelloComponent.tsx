import React from 'react'

interface Props {
    message: string
}

const MySimpleHelloComponent = (props: Props) => {
    const { message } = props;
    return (
        <h1>
            {message}
        </h1>
    )
}

export default MySimpleHelloComponent;
