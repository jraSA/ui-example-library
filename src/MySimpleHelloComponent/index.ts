import MySimpleHelloComponent from './MySimpleHelloComponent';

export default MySimpleHelloComponent;
