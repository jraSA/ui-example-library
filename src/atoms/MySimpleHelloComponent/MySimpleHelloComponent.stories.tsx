import React from 'react';
import { Meta } from '@storybook/react/types-6-0';
import MySimpleHelloComponent from '.';

export default {
    title: 'Atoms/MySimpleHelloComponent',
    component: MySimpleHelloComponent,
} as Meta;

const Template = (args) => <MySimpleHelloComponent {...args} />;

export const SimpleTextComponent = Template.bind({});
SimpleTextComponent.args = {
    message: 'foo-boo',
};
