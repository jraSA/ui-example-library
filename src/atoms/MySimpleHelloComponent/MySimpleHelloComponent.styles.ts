import styled from 'styled-components';
import BaseTheme from '../../theme/BaseTheme';
import { getThemeValue } from '../../theme/ThemeUtils';

const { color, fonts } = BaseTheme;

const { black } = color;
const { primaryFont } = fonts;

const Message = styled.div`
  font-family: ${primaryFont}, arial, sans-serif;
  text-align: left;
  color: ${(props) => getThemeValue(props, 'textColor', black)};
`;

export { Message };
