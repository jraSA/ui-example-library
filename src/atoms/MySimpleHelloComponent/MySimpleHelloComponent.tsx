import React from 'react';
import { Message } from './MySimpleHelloComponent.styles';

interface Props {
    message: string
};

const MySimpleHelloComponent = (props: Props) => {
    const { message } = props;
    return (
        <Message>
            {message}
        </Message>
    )
};

export default MySimpleHelloComponent;
