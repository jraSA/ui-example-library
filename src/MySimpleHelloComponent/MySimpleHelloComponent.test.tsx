import React from 'react';
import renderer from 'react-test-renderer';
import MySimpleHelloComponent from '../MySimpleHelloComponent';

describe('Render test', () => {
  const wrapper = renderer.create(<MySimpleHelloComponent message={'foo-boo'} />);

  test('WHEN I render the component THEN it should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
