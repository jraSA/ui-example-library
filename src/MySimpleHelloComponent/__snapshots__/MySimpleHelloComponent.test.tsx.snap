// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render test WHEN I render the component THEN it should match the snapshot 1`] = `
<h1>
  foo-boo
</h1>
`;
